crypto/evp/p_lib.c

```diff
@@ -896,6 +896,9 @@ int EVP_PKEY_type(int type)
 
 int EVP_PKEY_id(const EVP_PKEY *pkey)
 {
+    if (!evp_pkey_downgrade((EVP_PKEY *)pkey))
```

Comment:
Do we really need to downgrade to get the type??

Reply:
Not needed any more since 5e5bc836f so this has gone.